audio-echo/app/src/main/java/com/google/sample/echo/MainActivity.java

```diff
@@ -38,6 +46,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        //get run-time permission...
+        requestResourcePermission();
```

Comment:
when permission request is needed and denied, the status text box is going to show: "Error: permission not granted"  from then on click button will have no action at all. User needs to restart the app and grant the permission